update.py

```diff
@@ -28,17 +26,47 @@ def checkForUpdates(self):
             thread.daemon = True
             thread.start()
 
-        def worker(self):
+        def check_appcast(self) -> dict:
             import requests
             from xml.etree import ElementTree
 
-            r = requests.get(update_feed, timeout = 20, verify = (sys.version_info >= (2,7,9)))
-            feed = ElementTree.fromstring(r.text)
-            items = dict([(item.find('enclosure').attrib.get('{http://www.andymatuschak.org/xml-namespaces/sparkle}version'),
-                           item.find('title').text) for item in feed.findall('channel/item')])
-            lastversion = sorted(items, key=versioncmp)[-1]
-            if versioncmp(lastversion) > versioncmp(appversion):
-                self.root.nametowidget('.%s.%s' % (appname.lower(), 'status'))['text'] = items[lastversion] + ' is available'
+            newversion = None
+            try:
+                r = requests.get(update_feed, timeout=10)
+            except requests.RequestException as ex:
+                sys.stderr.write('Error retrieving update_feed file: {}\n'.format(str(ex)))
+            else:
+                try:
+                    feed = ElementTree.fromstring(r.text)
+                except SyntaxError as ex:
+                    sys.stderr.write('Syntax error in update_feed file: {}\n'.format(str(ex)))
+                else:
+
+                    items = dict()
```

Comment:
```suggestion
                    items = {}
```

We use literals elsewhere, so we should be consistent

Reply:
I think I prefer the more obvious (especially to any Python newbie) `dict()`.  Also PyCharm comes with a style check on by default to convert such literals *to* the constructor.  Not that PEP8 seems to mention it....

Yes, that would mean converting all the *others* to the constructor form.

Reply:
That works

Reply:
On the other hand, reviewing all those PyCharm options now I don't think I want the verboseness of always using dict(...) form, so yeah, literal it is.

Reply:
makes it easier to expand if we ever do so as well